utils/kubernetes/error.go

```diff
@@ -15,3 +22,23 @@ func ErrServiceDiscovery(err error) error {
 func ErrApplyHelmChart(err error) error {
 	return errors.NewDefault(errors.ErrApplyHelmChart, "Error applying helm chart: "+err.Error())
 }
+
+// ErrApplyHelmChart is the error which occurs in the process of applying helm chart
+func ErrNewKubeClient(err error) error {
```

Comment:
@kumarabd do you have an example of one of these also providing probable cause and suggested remediation?

Reply:
Same here, we're using a temp method which doesn't have cause and other properties. We can switch to that when we are final with the strategy.